sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java

```diff
@@ -122,11 +122,14 @@ public synchronized void start() {
             return;
         }
         logger.info("Starting a new event processor instance with id {}", this.identifier);
-        scheduler.set(Schedulers.newElastic("EventProcessor"));
+
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Out of curiousity, why the switch to an explicit ScheduledExecutorService over Schedulers.single() and [Scheduler.schedulePeriodically](https://projectreactor.io/docs/core/release/api/reactor/core/scheduler/Scheduler.html#schedulePeriodically-java.lang.Runnable-long-long-java.util.concurrent.TimeUnit-)?